### ISSUE TITLE
Make Rider EAP-compatible build of Cursive 

### ISSUE THREAD

**Issue by worrel** [cursive-ide/cursive#1488]
IntelliJ is now making a .Net-compatible version of IntelliJ in Project Rider.  #942 was closed because this wasn't the case previously.  Any chance we can get an EAP build of Cursive to test integration?

**Comment by Mechrophile** [comment]
[Link to the Priject Rider page for the lazy](https://www.jetbrains.com/rider/)

Looks awesome, and Cursive moving into the CLR would be so rad

**Comment by cursive-ide** [comment]
(pasting an answer from the mailing list here):

So it’s an interesting idea, and it should be possible but it won’t work out of the box. There’s probably a non-trivial amount of work to get it going, because the API for accessing C# classes will almost certainly be different than the one for accessing Java classes. Additionally, Rider is the first IntelliJ product (that I’m aware of) to not do all the symbol resolution itself but to delegate to a backend process - they do this so that they can share the implementation with ReSharper. 

Since Rider isn’t open source, figuring out how that works is likely to be difficult. I’d love to help with this, but realistically given the low adoption of ClojureCLR it’s unlikely to be a priority for me any time soon.

Sorry I don’t have a better answer.

**Comment by worrel** [comment]
Hey @cursive-ide no worries, I think as a first step we just wanted to see if we could get the vanilla Cursive plugin activated for Rider.  I cracked open the Cursive ZIP and updated the `until-build` to 163 & am now able to load the plugin from file.  However, it doesn't seem to activate at all - plug icon is grayed out in the plugins panel, and .clj files aren't associated (it actually suggests "La Clojure" for handling them 😞 ).

Can we leave this issue open for now to track this? https://www.jetbrains.com/rider/documentation/rider_survival_guide.html#work-in-progress suggests that "availability of IntelliJ platform plugins" is a work in progress, implying they intend to make it happen at some point.

**Comment by cursive-ide** [comment]
Yes, definitely leave this issue open - hopefully they'll publish some information at some point about how to integrate with Rider.

**Comment by Mechrophile** [comment]
Awesome. good news that you're willing to keep it open!  I'd bet anything that CursiveCLR would go a long way to pushing the adoption of clojure CLR.  Its like building roads -- if there arent any, people aren't inclined to adopt cars as a transportation model.

Cant wait for rider to get developed a bit more.

**Comment by BorisKourt** [comment]
Replying 1000 years later. Is this something that might still be possible or is Rider too far off the radar?

**Comment by cursive-ide** [comment]
Rider is too far off the radar, sorry - it would be a huge change due to the architectural differences in Rider, and given the low adoption of ClojureCLR this is very unlikely to get any priority.
